lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -76,6 +77,20 @@ data TransactionLayer t k = TransactionLayer
         -- This expects as a first argument a mean to compute or lookup private
         -- key corresponding to a particular address.
 
+    , mkDelegationCertTx
+        :: PoolId
+        -> (k 'AddressK XPrv, Passphrase "encryption")
+        -> (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+        -> [(TxIn, TxOut)]
+        -> [TxOut]
+        -> Either ErrMkStdTx (Tx, SealedTx)
```

Comment:
>  Plus, this function is likely to have different errors, or, errors that could be interpreted differently.

But it doesn't? What errors are you seeing it having?

And how would be the preferred way of handling the duplication of constructors then?
```haskell
-- | Possible signing error
newtype ErrMkStdTx
    = ErrKeyNotFoundForAddress Address
    -- ^ We tried to sign a transaction with inputs that are unknown to us?
    deriving (Eq, Show)

-- | Possible signing error
newtype ErrMkDelegationCertTx
    = ErrKeyNotFoundForAddress Address
    -- ^ We tried to sign a transaction with inputs that are unknown to us?
    deriving (Eq, Show)
```

Reply:
Then, if errors are indeed the same, we should rename `ErrMkStdTx` to not suggests that it refers to `mkStdTx` but that it's a more general type of error. What about:

```hs
newtype ErrInputNotOwned 
    = ErrKeyNotFoundForAddress Address
```

Reply:
> mkStdTx but that it's a more general type of error

What about simply:
```haskell
newtype ErrMkTx
    = ErrKeyNotFoundForAddress Address
```
which is used both by `mkStdTx`, and `mkDelegationCertTx`?

Reply:
As you prefer :)

Reply:
Both have their qualities. I think `ErrInputNotOwned` is neat 🤷‍♂ 

But I'm defaulting to `ErrMkTx` because it is more consistent with the original `ErrMkStdTx` named after the _operation_ rather than _what could go wrong_.